### PR TITLE
Improve HTML GUI Toolbar

### PR DESCRIPTION
**Pull request by m273d15**
This PR:
* enhances the toolbar buttons:
  * Disables the toolbar buttons (add contact, share project) if not connected to an XMPP account
  * Make the dropdown list of the split-button clickable (change current account by clicking on an account in the list)
* contains some minor refactorings
* adds tests of the toolbar (and contained components)

### CODE REVIEW

File: ui.frontend/html/src/components/ToolBar/ConnectionSplitButton.jsx [hunk 1]
```diff
@@ -1,32 +1,49 @@
...
+    ))}
+  </div>
+)
 
 class ConnectionSplitButton extends React.Component {
-  constructor (props) {
+  constructor(props) {
     super(props)
+    const onConnect = this.props.core.doConnect
+    const onDisconnect = this.props.core.doDisconnect
```
Comment (Drakulix) [inline ui.frontend/html/src/components/ToolBar/ConnectionSplitButton.jsx]: Not important, but you could also do:
`const { doConnect, doDisconnect } = this.props.core`

File: ui.frontend/html/test/connectionSplitButton.test.jsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+
+describe('<ConnectionSplitButton />', () => {
+
+
+    function createEmptyStore() {
+        return new SarosStoreBuilder().build()
+    }
+
+    itRendersWithoutExploding(mount(<ConnectionSplitButton core={createEmptyStore()} />).find('ConnectionSplitButton'))
```
Comment (Drakulix) [inline ui.frontend/html/test/connectionSplitButton.test.jsx]: :heart:

File: ui.frontend/html/src/components/ToolBar/ConnectionSplitButton.jsx [hunk 3]
```diff
@@ -1,32 +1,49 @@
...
-      'CONNECTED': { onClick: this.props.onDisconnect, messageId: 'action.disconnect', icon: images.accountConnectedIcon },
-      'ERROR': { onClick: noop, messageId: 'action.connectionError', icon: images.accountConnectionErrorIcon },
-      'CONNECTING': { onClick: noop, messageId: 'action.connecting', icon: images.accountConnectingIcon },
-      'DISCONNECTING': { onClick: noop, messageId: 'action.disconnecting', icon: images.accountDisconnectingIcon }
-    }
+    const onConnect = this.props.core.doConnect
+    const onDisconnect = this.props.core.doDisconnect
+
+    this.connectionStateMap = {}
+    this.connectionStateMap[connectionStates.INITIALIZING] = { onClick: onConnect, messageId: 'action.connect', icon: images.accountDisconnectedIcon }
```
Comment (srossbach) [inline ui.frontend/html/src/components/ToolBar/ConnectionSplitButton.jsx]: What is this state ? Looking at https://github.com/saros-project/saros/blob/master/core/src/saros/net/ConnectionState.java there is no such state.

Comment (m273d15) [reply ui.frontend/html/src/components/ToolBar/ConnectionSplitButton.jsx]: I think this state was introduced in order to define the state:
HTML-GUI is started, but no state was send/received.

**Comment by m273d15** [comment]
See #682

**Pull request closed without merging**